sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs

```diff
@@ -50,7 +50,7 @@ public AzureMonitorTransmitter(AzureMonitorExporterOptions options, IPlatform pl
 
             if (_fileBlobProvider != null)
             {
-                _transmitFromStorageHandler = new TransmitFromStorageHandler(_applicationInsightsRestClient, _fileBlobProvider, _transmissionStateManager);
+                _transmitFromStorageHandler = new TransmitFromStorageHandler(_applicationInsightsRestClient, _fileBlobProvider, _transmissionStateManager, _connectionVars);
```

Comment:
Should we add few properties to `ApplicationInsightsRestClient` instead of passing additional args in many places?

Reply:
I looked for alternatives to this but nothing stood out to me.

When our RestClient creates the `HttpMessage`, there is a property bag that we could add these values.
This wouldn't affect what is sent to ingestion, the `HttpMessage` is a wrapper around the `Request` and `Response` objects. But I'm not sure how this would affect memory.

Reply:
Tested `HttpMessage`. This is the better approach. Will make the change